protocol/binary_test.go

```diff
@@ -297,6 +310,7 @@ func TestBinaryEOFFailure(t *testing.T) {
 		{},
 		{0x00}, // incomplete length
 		{0x00, 0x00, 0x00, 0x01}, // length mismatch
+		{0x22, 0x6e, 0x6f, 0x74}, // really long length
```

Comment:
That will involve changing all `checkEOFError`-based tests. I would prefer to hold off on this change until this becomes an actual issue. Usually there are only 1-2 cases for each test based on that.